lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -177,25 +188,36 @@ class AwsCompileCloudFrontEvents {
                 ? event.cloudFront.pathPattern
                 : undefined;
 
-            let origin = createOrigin(functionName, event.cloudFront.origin, this.provider.naming);
-
-            const existingOrigin = _.find(
-              origins,
-              o => o.DomainName === origin.DomainName && o.OriginPath === origin.OriginPath
-            );
+            let origin = createOrigin(event.cloudFront.origin, this.provider.naming);
+            const existingOrigin = _.find(origins, o => o.Id === origin.Id);
 
             if (!existingOrigin) {
               origins.push(origin);
             } else {
               origin = extendDeep(existingOrigin, origin);
             }
 
+            if (event.cloudFront.defaultOrigin === true) {
+              if (defaultOrigin !== undefined && defaultOrigin !== origin) {
```

Comment:
Let's make it `defaultOrigin && defaultOrigin !== origin` (otherwise values as `null` are treated as if `defaultOrigin` is set)